### PR TITLE
Build and deploy docker images on CI

### PR DESCRIPTION
**Pull request by lpil**
blow-ring

CI IAM user has the EB full access policy + this inline policy:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "elasticbeanstalk:CreateApplicationVersion",
                "elasticbeanstalk:DescribeApplicationVersions"
            ],
            "Resource": [
                "arn:aws:elasticbeanstalk:*:*:applicationversion/react-london/*",
                "arn:aws:elasticbeanstalk:*:*:environment/react-london/*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "s3:ListBucket"
            ],
            "Resource": [
                "arn:aws:s3:::*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "s3:PutObject",
                "s3:GetObject",
                "s3:GetObjectAcl"
            ],
            "Resource": [
                "arn:aws:s3:::elasticbeanstalk-*",
                "arn:aws:s3:::elasticbeanstalk-*/*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "ecr:GetAuthorizationToken"
            ],
            "Resource": "*"
        },
        {
            "Effect": "Allow",
            "Action": [
                "ecr:InitiateLayerUpload",
                "ecr:UploadLayerPart",
                "ecr:CompleteLayerUpload",
                "ecr:BatchCheckLayerAvailability",
                "ecr:PutImage"
            ],
            "Resource": [
                "arn:aws:ecr:*:*:repository/react-london"
            ]
        }
    ]
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,14 +32,21 @@ make lint
...
+Beanstalk Application Versions section of the AWS web console, or with the
+commands below:
+
 ```sh
-# Deploy staging
+# Deploy the current commit to staging
 make deploy-staging
-# Deploy production
+# Deploy the current commit to production
 make deploy-production
```
Comment (asavin) [inline README.md]: Does this mean that we're building new Docker image when running `make deploy-production`?

Comment (lpil) [inline README.md]: No, it's just looking for a version called `$BRANCH-$SHA` that's already registered and then telling elastic beanstalk to deploy that.

Comment (asavin) [inline README.md]: Can we have CircleCI do this on new tag push?

Comment (lpil) [inline README.md]: Sure, seems like another story to me though

**Pull request merged**